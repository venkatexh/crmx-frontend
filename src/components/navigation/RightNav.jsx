import "../../sass/components/navigation.component.scss";
import { useDispatch, useSelector } from "react-redux";
import ProfileTab from "./ProfileTab";
import { useState } from "react";
import NotificationTab from "./NotificationTab";
import { setNewNotification } from "../../redux/actions/notifications/setNewNotification";
import { Link } from "react-router-dom";

const RightNav = () => {
  const state = useSelector(({ loggedUser, newNotification }) => ({
    loggedUser,
    newNotification,
  }));
  const [tabOpen, setTabOpen] = useState(false);
  const [tab, setTab] = useState("");
  const dispatch = useDispatch();

  const tabToRender = () => {
    if (tab === "profile") {
      return <ProfileTab />;
    } else if (tab === "notification") {
      return <NotificationTab />;
    }
  };

  return (
    <div className={"rightNav"}>
      {tabOpen ? <div>{tabToRender()}</div> : <></>}
      <div
        className={"profileBtn"}
        onClick={() => {
          setTab("profile");
          setTabOpen(!tabOpen);
        }}
      >
        {state.loggedUser.firstName[0]}
      </div>
      <div
        className={"iconContainer"}
        onClick={() => {
          setTab("notification");
          setTabOpen(!tabOpen);
          dispatch(setNewNotification(false));
        }}
      >
        <img src={"/icons/bell.png"} alt={"icon"} style={{ height: "32px" }} />
        {state.newNotification ? (
          <div className={"notificationSymbol"}>&#8226;</div>
        ) : null}
      </div>
      <Link to={"/organization-settings"} className={"iconContainer"}>
        <img
          src={"/icons/setting.png"}
          alt={"icon"}
          style={{ height: "32px" }}
        />
      </Link>
      <div className={"iconContainer"}>
        <img
          src={"/icons/magic-wand.png"}
          alt={"icon"}
          style={{ height: "32px" }}
        />
      </div>
    </div>
  );
};

export default RightNav;
