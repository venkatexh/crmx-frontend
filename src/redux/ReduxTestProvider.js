import { Provider } from "react-redux";

const ReduxTestProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
);

export default ReduxTestProvider;
