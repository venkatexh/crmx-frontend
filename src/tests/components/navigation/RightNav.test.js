import { shallow } from "enzyme";
import React from "react";
import RightNav from "../../../components/navigation/RightNav";
import ReduxTestProvider from "../../../redux/ReduxTestProvider";

it("Right nav component snapshot", () => {
  expect(
    shallow(
      <ReduxTestProvider>
        <RightNav />
      </ReduxTestProvider>
    )
  ).toMatchSnapshot();
});
