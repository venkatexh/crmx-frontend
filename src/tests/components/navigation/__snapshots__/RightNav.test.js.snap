// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Right nav component snapshot 1`] = `ShallowWrapper {}`;
